Add resources and data sources for Prism Central backup, restore, deployment and unregistration using v4

Haroon-Dweikat-Ntx opened the pull request:
Prism V2 resources and data sources that supports these operations:
- PC Deploy 
- PC Unregistration 
- PC  Backup Target(CRUD)
- PC Restore Source
- List Restorable Domain Managers
- Get or List PC Restore Points
- PC Restore

Haroon-Dweikat-Ntx committed:
fix category service update

Haroon-Dweikat-Ntx committed:
fix iamv2 acc test

Haroon-Dweikat-Ntx committed:
fix storage acc test

Haroon-Dweikat-Ntx committed:
V4 volumes on v4-temp-design-2.0 (#25)

* address groups v4

* service groups

* resource for service groups

* crud for service groups

* CRUD for address groups

* data source for network security

* CRUD for network security

* microseg sdk pointing to internals

* datasource for directory services

* CRUD for directory service

* datasource for saml

* CRUD for idp

* delete Operation for directory service

* CRUD for user groups

* datasource for categories

* Crud and tcs for categories

* crud & test for images

* sdk versioning

* templates datasource

* datasource for template versions

* deploy templates

* spec for vms

* info for volume Groups

* add targetPrefix atrribute to volume group/s schema

* create Ops

* CRUD for volume group, Change the names of imported packages to more declarative names.

* CUD ops done

* Get VMs

* fix update for Volume group, working on volume disk -> info done, CRUD there is a bug on update func  CRD works as expected

* VMs

* CRUD for vm disks

* volume group disk CRUD done, fix schema attribute for Vg

* CRUD/ds serial ports

* cdrom CRUD

* insert/eject cdrom

* vm actions power

* List available iscsi clients + Attach/Detach iscsi client to vg

* merge conflict

* Revert "merge conflict"

This reverts commit 7b8edb9e177ed96b19d2d8e0fde52e37647502e8.

* List all the category details that are associated with the Volume Group.

* Attach/Detach an AHV VM from the given Volume Group.

* Iscsi Clients Provider, Fetch an iSCSI client details, List all the iSCSI clients.

* revert pushed unwanted changes

* revert pushed unwanted changes

* acc test for volume group data source and resource

* acceptance test for volume group data source

* volume group data source acctest is done

* acceptance test for volume resource modules, ->  acceptance test done

* Add Description for schema attribute's

* List all the VM attachments for a Volume Group INFO -> Development + Test

* Volume Group examples

* Doc for Volume groups & Attach/Dettach vg to vm

* Docs for Volume Group vm, volume group  iscsi client, available  iscsi clients, Volume Group disks and volume group category info

* refactor volumes test cases to use test_config_v4.json

* revert pushed unwanted changes

* add test_config_v4.json file

* refactor volumes test cases to use test_config_v4.json

* rename volume module from v4 to v2

* Refactor acceptance test cases to be more dynamic and reduce dependency on the JSON configuration file

* rename volume package to volumesv2

* remove other modules

* add examples

* add examples

* ignore vendor

---------

Co-authored-by: Abhishek <abhishek.chaudhary@nutanix.com>

Haroon-Dweikat-Ntx committed:
V4 data protection on v4-temp-design (#31)

* Feat/v4 recovery points on v4-recovery-points (#26)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* Feat/v4 replicate recovery points on v4 recovery points (#27)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* implement recovery point replicate in progress

* implementation of recovery point replication

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* implement acc test for recovery point replication

* add examples for recovery point replication

* add docs for recovery point replication

* add replicated_rp_ext_id attribute to store the replicated recovery point uuid

* Feat/v4 recovery points on v4 data protection (#28)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix comments on the added DiffSub function

* Feat/v4 recovery point restore on v4-data-protection (#29)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* implement recovery point restore resource

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix comments on the added DiffSub function

* remove commented code

* acc tests for recovery point restore

* examples for recovery point restore

* examples for recovery point restore

* docs for recovery point restore

* Feat/v4 vm recovery point info  (#30)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* implement recovery point restore resource

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix comments on the added DiffSub function

* remove commented code

* acc tests for recovery point restore

* examples for recovery point restore

* examples for recovery point restore

* docs for recovery point restore

* implement vm recovery point info

* Acc tests for vm recovery point info

* examples for vm recovery point info

* docs for vm recovery point info

* fix links schema

Haroon-Dweikat-Ntx committed:
Feat/v4 static routes on v4 temp design (#32)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* Feat/v4 recovery points on v4-recovery-points (#26)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* Feat/v4 replicate recovery points on v4 recovery points (#27)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* implement recovery point replicate in progress

* implementation of recovery point replication

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* implement acc test for recovery point replication

* add examples for recovery point replication

* add docs for recovery point replication

* add replicated_rp_ext_id attribute to store the replicated recovery point uuid

* implement recovery point restore resource

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix comments on the added DiffSub function

* remove commented code

* acc tests for recovery point restore

* examples for recovery point restore

* examples for recovery point restore

* docs for recovery point restore

* implement vm recovery point info

* Acc tests for vm recovery point info

* examples for vm recovery point info

* docs for vm recovery point info

* implement vm revert logic

* acc test for vm revert logic

* acc test for vm revert logic

* examples for vm revert logic

* docs for vm revert logic

* CRUD for routes

* info for Route and route tables

* acc tests for routes and route tables

* examples for routes and route tables

* docs for routes and route tables

* update examples for routes and route tables

* fix links schema

abhimutant committed:
etag value to pointers

Haroon-Dweikat-Ntx committed:
V4 vmm   (#34)

* address groups v4

* service groups

* resource for service groups

* crud for service groups

* CRUD for address groups

* data source for network security

* CRUD for network security

* microseg sdk pointing to internals

* datasource for directory services

* CRUD for directory service

* datasource for saml

* CRUD for idp

* delete Operation for directory service

* CRUD for user groups

* datasource for categories

* Crud and tcs for categories

* crud & test for images

* sdk versioning

* templates datasource

* datasource for template versions

* deploy templates

* spec for vms

* create Ops

* CUD ops done

* Get VMs

* VMs

* CRUD for vm disks

* CRUD/ds serial ports

* cdrom CRUD

* insert/eject cdrom

* vm actions power

* vm shutdown actions

* CRUD gpus

* associate/diassociate categories to VM

* missing return in vms

* tcs for vm resource

* tcs for vms

* acc for images

* adding more tcs

* Vms disk Tcs

* NGT install/Uninstall Resource

* tcs for serial port

* resource tcs for cdrom

* TCs for cdroms

* tcs for vm power state

* power state testcase

* shutdown Tcs

* Adding TCs for Gpus

* fixing error logic

* template versions testcases

* template deploy testcase

* template actions

* docs

* Docs for images

* docs for vm disks

* docs for serial ports

* docs for cdroms

* testcases and docs for insert eject cdrom

* docs for vm power action

* docs for shutdown vm

* docs for gpus

* vm categories docs and tcs

* docs for templates crud

* docs for template version

* docs for template guest os

* change sdk from ntnx-api-golang-sdk-internal to ntnx-api-golang-clients

* change sdk from ntnx-api-golang-sdk-internal to ntnx-api-golang-clients

* NGT Install/Uninstall on Vm implementation done, get Vm NGT configuration implementation done

* acc tests for NGT installations,

* test done, update test must be verified

* docs for NGT

* ngt_upgrade module

* test config v4 file

* fix import to use new Sdk "ntnx-api-golang-clients" insted of internal sdk

* test cases for NGT installation + update + ngt info done , docs done

* remove unused func

* dev for nutanix_ngt_insert_iso_v4

* docs for nutanix_ngt_insert_iso_v4

* ngt insert tools acc tests done

* validation on template CRUD schema, change resource name from v4 to v2

* validation on template CRUD expanders,

* fix kvPairs values, now value support values for :integer,list array of integer, array of list, map and object,
testing template create

* fix schema for template data source

* validation on get template logic

* change template data source name from v4 to v2

* add acc tests template data source

* add acc tests template data source

* add acc tests template resource

* add examples for template CRUD + info

* fix docs for template

* template deploy : change name from v4 to v2, add examples, fix acc test

* add more validations on template create and update, creating a template is only allowed using vm reference, updating can use template_version_reference or template_vm_reference, only one of them.
make version_id Optional and set the default value to the latest version of the template when it's not provided in the configuration
add more examples.
version_name and version_description both required when updating the template.

* merging VMs entities ( disks, nics, serial ports, gpu, cdrom)

* fix sdk's, use internal sdk

* create vm and relative resources during test, refactor ngt installation dev

* fix ngt dev and acc tests

* ngt examples

* change name from v4 to v2

* add `is_scsi_controller_enabled` & `is_cpu_hotplug_enabled` to vmm module

* categories attach logic

* cdrom insert eject resource

* fixing merging issues

* merge vm clone, change vm name from v4 to v2

* change nutanix_virtual_machine_v4 from v4 to v2 in test cases files

* change nutanix_virtual_machine_v4 from v4 to v2 in test cases files

* merge vm clone, change vm name from v4 to v2

* unified image resource in branch

* change subnets datasource used in test files from v4 to v2

* vm clone resource release ready

* add network device resource, assign/ remove ip address resource [includes test cases examples and docs]

* add network device datasource

* add migrate device resource with examples and docs

* change names from v4 to v2

* add guest customization resources with docs and examples

* add image placement module with docs and examples

* add ipv4_info to vmm resource schema, update sdk's to the latest

* fix image police placement, fix e-tage header so that now it passed as a pointer not as a value.

* fix images test cases,

* fix e-tage header so that now it passed as a pointer not as a value.

* set default power state for vm as ON

* fix vmm tests and run all tests on RC3 setup

* fix acc tests and Run all vmm tests on RC3 setup

* remove vendor

* change package name from vmm to vmmv2

---------

Co-authored-by: Abhishek <abhishek.chaudhary@nutanix.com>
Co-authored-by: Pradeepsingh Bhati <pradeep.bhati@nutanix.com>

Haroon-Dweikat-Ntx committed:
V4 clusters managment (#33)

* datasource for pbrs

* lint fixes. go error (gomnd, gosimple, golint)

* go checks, magic numbers(gomnd)

* fix config testcase as base client will differ in sdks

* datadource for clusters

* lint fixes

* host datasource

* lint fixes

* address groups v4

* service groups

* resource for service groups

* crud for service groups

* CRUD for address groups

* data source for network security

* CRUD for network security

* microseg sdk pointing to internals

* datasource for directory services

* CRUD for directory service

* datasource for saml

* CRUD for idp

* delete Operation for directory service

* CRUD for user groups

* datasource for categories

* Crud and tcs for categories

* crud & test for images

* sdk versioning

* templates datasource

* datasource for template versions

* deploy templates

* spec for vms

* create Ops

* CUD ops done

* Get VMs

* VMs

* CRUD for vm disks

* CRUD/ds serial ports

* cdrom CRUD

* insert/eject cdrom

* vm actions power

* vm shutdown actions

* CRUD gpus

* missing return in vms

* vm clone resource

* tcs for vm resource

* tcs for vms

* acc for images

* adding more tcs

* Vms disk Tcs

* tcs for serial port

* resource tcs for cdrom

* TCs for cdroms

* vm clone example and docs

* tcs for vm power state

* power state testcase

* shutdown Tcs

* Adding TCs for Gpus

* vm clone testcases

* fic for argument naming

* resource and tests for update guest customization for next boot

* guest customization update doc and example

* acc test for data_source host v2

* docs for data source hosts v4

* cluster v4 resource

* fix for b2 version

* fix for version

* resources and data sources for (SNMP) user, trap and transport port

* add, remove node from cluster resources

* prism sdk version

* schema for Discover Unconfigured Nodes module

* fetch unconfigured nodes network datasource

* change Discover unconfigured Nodes module from resource to data source

* fix for version

* resources and data sources for (SNMP) user, trap and transport port

* add, remove node from cluster resources

* prism sdk version

* fetch unconfigured nodes network datasource

* change Discover unconfigured Nodes module from resource to data source

* Discover Unconfigured Nodes dev almost done, waiting a pc to testing it

* remove duplicate "nutanix_discover_unconfigured_nodes_v4" key in datasource map

* docs and fixes

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* Feat/v4 recovery points on v4-recovery-points (#26)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* Feat/v4 replicate recovery points on v4 recovery points (#27)

* add sdk, create client for Data Protection and add Recovery Point Api Client

* implement get recovery point data source

* implement list recovery points data source

* implement CRUD recovery points resource, update api is not supported for RP

* implement implement recovery points resource acc tests, in progress

* recovery points datasource acc tests.

* recovery points examples

* recovery points docs

* create Vg within RP test cases

* fix provider version on examples

* implement recovery point replicate in progress

* implementation of recovery point replication

* add test config on test_config_v2.json file

* fix acc test when create vg, remove unnecessary filepath parameter, remove vm uuid from test_config_v2.json file

* implement acc test for recovery point replication

* add examples for recovery point replication

* add docs for recovery point replication

* add replicated_rp_ext_id attribute to store the replicated recovery point uuid

* implement recovery point restore resource

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix terraform plan for recovery point, there is no update in the resource just changing in the order of vm_recovery_points and volume_group_recovery_points array

* fix comments on the added DiffSub function

* remove commented code

* acc tests for recovery point restore

* examples for recovery point restore

* examples for recovery point restore

* docs for recovery point restore

* implement vm recovery point info

* Acc tests for vm recovery point info

* examples for vm recovery point info

* docs for vm recovery point info

* implement vm revert logic

* acc test for vm revert logic

* acc test for vm revert logic

* examples for vm revert logic

* docs for vm revert logic

* CRUD for routes

* info for Route and route tables

* acc tests for routes and route tables

* examples for routes and route tables

* docs for routes and route tables

* update examples for routes and route tables

* remove old packages

* change clusters sdk from external sdk to internal sdk

* change function names from v4 to v2, fix schema and function params

* change package name from clusters to clustersv2

* change resources and datasource name from v4 to v2

* acc tests for datasource clusters

* fix bugs in resource Cluster implementation and schema

* acc tests for resource Cluster

* fix test_config_v2.json file

* add missing "categories" attribute to cluster schema

* fix resource cluster docs and add missed attribute http_proxy_list and http_proxy_white_list to cluster schema

* fix datasource cluster module, add missed attribute http_proxy_list and http_proxy_white_list, current_cluster_fault_tolerance, desired_cluster_fault_tolerance, redundancy_status, categories, cluster_profile_ext_id,  backup_eligibility_score to cluster schema

* fix datasource cluster/ clusters  docs

* add clusters examples

* add tests and docs for nutanix_clusters_unconfigured_node_networks_v2 and nutanix_clusters_discover_unconfigured_nodes_v2

* docs & examples for nutanix_cluster_remove_node_v2 and nutanix_cluster_add_node_v2

* implement PE-PC registration resource {Create & Delete}

* implement PE-PC registration datasource info

* acc tests for PE-PC registration resource

* acc tests for PE-PC registration datasource

* docs and examples for PE-PC registration datasource and resource

* examples for nutanix_host_v2, change then name from v4 to v2

* change the name of the following resources :
nutanix_cluster_pc_entity_v2 -> nutanix_pe_pc_entity_v2
nutanix_cluster_pc_entities_v2 -> nutanix_pe_pc_entities_v2
nutanix_cluster_pc_register_v2 -> nutanix_pe_pc_registration_v2

* add unconfigured nodes nodes schema

* acc test for
unconfigured nodes nodes schema

* update docs and examples for unconfigured nodes

* update schema (add missing `is_secure_booted property`) and test_config_v2 for unconfigured nodes

* fix cluster v2 schema and create logic, add test cases and examples and update docs

* fix cluster v2 schema and create logic, add test cases and examples and update docs

* update pe-pc registration schema, test cases, examples, update docs

* pc fetch and list will be future work

* Adding dev and test for cluster CRUD module, changes discover unconfigured nodes from datasource to resource

* Adding dev for fetch network info for un configured node,

* Adding dev for fetch network info for un configured node,

* fix dev for remove node from a cluster ,

* fix dev for cluster CRUD, build info, upgrade_status and current_cluster_fault_tolerance is computed attributes ,

* update docs & examples for fetch network info for un configured node

* update docs & examples for fetch network info for un configured node

* acc test for add node to cluster, its cover fetch network info for unconfigured node, discover unconfigured node and discover unconfigured cluster node and add node to cluster

* remove validation for current_cluster_fault_tolerance because its computed attribute

* merge and node and remove node resource in a single resource "add node"

* fix the add/remove acc test case

* fix vendor

* fix vendor step1: delete vendor

* fix vendor step1: add vendor from temp design branch

* Remove vendor from the repository

* add vendor to the repository

* Remove vendor from the repository

* add vendor to the repository

* fix cluster management tests, fix etag, so now it passed as a pointer not as value

---------

Co-authored-by: Abhishek <abhishek.chaudhary@nutanix.com>
Co-authored-by: Gevorg <gevorg.khachatryan@nutanix.com>

Haroon-Dweikat-Ntx committed:
REMOVE test_config_v4.json

Haroon-Dweikat-Ntx committed:
REMOVE test_config_v4.json

Haroon-Dweikat-Ntx committed:
cluster acc test passed on RC3 setup

abhimutant committed:
waiting for IP to be assigned

Haroon-Dweikat-Ntx committed:
fix data protection test, run tests on RC3 setup

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/v4-temp-design-2.0' into v4-temp-design-2.0

abhimutant committed:
deleted the unified resources from vmm

Haroon-Dweikat-Ntx committed:
change vmm resource name from v4 to v2

Haroon-Dweikat-Ntx committed:
fix insert/eject test case

Haroon-Dweikat-Ntx committed:
add nutanix_vm_shutdown_action_v2 resource

Haroon-Dweikat-Ntx committed:
fix vm gpus logic and run test, test is passed

abhimutant committed:
unified datasources for vm module

Haroon-Dweikat-Ntx committed:
remove commented lines, change data source used to fetch cluster to nutanix_clusters_v2

Haroon-Dweikat-Ntx committed:
fix Iam test cases, run Iam on RC3

Haroon-Dweikat-Ntx committed:
fix dataprotection test cases, run dataprotection on RC3

Haroon-Dweikat-Ntx committed:
fix networking test cases, run networking  on RC3

Haroon-Dweikat-Ntx committed:
fix vmm test cases, run vmm on RC3

Haroon-Dweikat-Ntx committed:
fix storage containers test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix categories test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix storage containers test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix networking test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix volumes test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix volumes test cases, run on RC3

Haroon-Dweikat-Ntx committed:
fix ipV4Config in vm create

Haroon-Dweikat-Ntx committed:
changed print statements to log, refactor test case

Haroon-Dweikat-Ntx committed:
change print statements to log, fix FloatingIPv2ConfigWithVMNic test case

Haroon-Dweikat-Ntx committed:
use nutanix_clusters_v2, fix template actions resource, increase the timeout for waiting to assign ip address to vm, remove template version resource

Haroon-Dweikat-Ntx committed:
add acc tests for operations datasource, fix other test cases

Haroon-Dweikat-Ntx committed:
fix vm names created during the test case

abhimutant committed:
handling guestCustomisation for custom key value pairs

Haroon-Dweikat-Ntx committed:
add multi_env_isolation_rule_spec to network security rule

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/v4-temp-design-2.0' into v4-temp-design-2.0

Haroon-Dweikat-Ntx committed:
add multi_env_isolation_rule_spec to network security rule

Haroon-Dweikat-Ntx committed:
fix creating Role and acp, uses only operations of type Create

Haroon-Dweikat-Ntx committed:
in v4.0 we have name, expirationTime, status, recoveryPointType inside vm recovery points

Haroon-Dweikat-Ntx committed:
separate schema For Guest Customization between vm and template, because its little different

Haroon-Dweikat-Ntx committed:
Fix examples for recovery points and acp

Haroon-Dweikat-Ntx committed:
change project structure

Haroon-Dweikat-Ntx committed:
add more examples, fix typo in examples and docs

Haroon-Dweikat-Ntx committed:
remove vm cdrom, disks, gpu, power, categories and serial port docs

Haroon-Dweikat-Ntx committed:
restructure v2 modules, now its under services directly

Haroon-Dweikat-Ntx committed:
test_config_v2 template file

Haroon-Dweikat-Ntx committed:
exclude vendor, bin folder

Haroon-Dweikat-Ntx committed:
remove cluster snmp resource

abhimutant committed:
adding v4 resource datasource in readme

abhimutant committed:
changing the SDKs from internal to GA SDKs

abhimutant committed:
commenting the params for GA release

abhimutant committed:
fixing lint

abhimutant committed:
lint fixes

abhimutant committed:
lint fixes

abhimutant committed:
skip gocritic lint for ifElseChain

abhimutant committed:
fixing lint issues

abhimutant committed:
fixing all lint from cli

abhimutant committed:
gofmt

abhimutant committed:
golint fixes

abhimutant committed:
fixing golint and unnecessary conversion

abhimutant committed:
golint fixes

abhimutant committed:
fix

abhimutant committed:
gomnd fix

abhimutant committed:
fixing testname

abhimutant committed:
markdown spelling mistake

Haroon-Dweikat-Ntx committed:
V4 design 2.0 sdk fixes (#700)

abhimutant committed:
adding runners configurations

abhimutant committed:
dummy run

Haroon-Dweikat-Ntx committed:
Deploy PC Dev [Create PC and Read PC Configuration]

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
fix networking test cases name from TestAccNutanix to TestAccV2Nutanix

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

abhimutant committed:
adding check for cluster resource test cases

abhimutant committed:
Merge branch 'v4-temp-design-2.0-structure' of github.com:nutanix/terraform-provider-nutanix into v4-temp-design-2.0-structure

Haroon-Dweikat-Ntx committed:
fix TestAccV2NutanixUsersResource_LDAPUser, delete user using V3 api to automate the test

Haroon-Dweikat-Ntx committed:
fix TestAccV2NutanixClusterAddNodeResource_Basic, we need four nodes to for this test case

Haroon-Dweikat-Ntx committed:
remove unnecessary check in pc registration test cases

abhimutant committed:
Merge branch 'master' of github.com:nutanix/terraform-provider-nutanix into v4-temp-design-2.0-structure

abhimutant committed:
fixing goimport error

Haroon-Dweikat-Ntx committed:
rename vmmv2 test case to TestAccV2Nutanix...

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/v4-temp-design-2.0-structure' into v4-temp-design-2.0-structure

abhimutant committed:
debugging pipeline

abhimutant committed:
Merge branch 'v4-temp-design-2.0-structure' of github.com:nutanix/terraform-provider-nutanix into v4-temp-design-2.0-structure

abhimutant committed:
adding lines to debug pipeline issue

Haroon-Dweikat-Ntx committed:
fix cluster entity test cases

Haroon-Dweikat-Ntx committed:
Merge branch 'v4-temp-design-2.0-structure' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
unregister cluster dev

Haroon-Dweikat-Ntx committed:
Backup Target CRUD Development

Haroon-Dweikat-Ntx committed:
fetch and list pc configuration details data sources dev

Haroon-Dweikat-Ntx committed:
fetch and list pc configuration details data sources dev

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
restore source CRUD dev

Haroon-Dweikat-Ntx committed:
add "PENDING" to the task state list

Haroon-Dweikat-Ntx committed:
restore source info dev

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
Merge branch 'feat/2.1/backup-target' into feat/2.1/restore-pc

Haroon-Dweikat-Ntx committed:
Merge branch 'feat/2.1/restore-source' into feat/2.1/restore-pc

Haroon-Dweikat-Ntx committed:
merge deploy pc branch

Haroon-Dweikat-Ntx committed:
restore pc module dev

Haroon-Dweikat-Ntx committed:
add to [num_vcpus, memory_size_bytes, data_disk_size_bytes] resource_config for deploy pc

Haroon-Dweikat-Ntx committed:
restore point info dev

Haroon-Dweikat-Ntx committed:
add restore point and restore pc to provider

Haroon-Dweikat-Ntx committed:
unregister cluster pc dev , move register pc module under prism package

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
merge other prism branches

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
acc tests for deploy pc, backup target, restore a source, restore pc and unregister cluster inside one file resource_nutanix_deploy_pc_v2_test.go, the test case involved testing related resources and datasource's

Haroon-Dweikat-Ntx committed:
deploy pc resource docs and pc details data sources docs

Haroon-Dweikat-Ntx committed:
deploy pc resource docs and pc details data sources docs

Haroon-Dweikat-Ntx committed:
backup target resource and data sources docs

Haroon-Dweikat-Ntx committed:
restore source resource and data sources docs

Haroon-Dweikat-Ntx committed:
restore pc resource docs

Haroon-Dweikat-Ntx committed:
unregister cluster  resource docs

Haroon-Dweikat-Ntx committed:
unregister cluster example

Haroon-Dweikat-Ntx committed:
restore pc example

Haroon-Dweikat-Ntx committed:
restore source example

Haroon-Dweikat-Ntx committed:
backup target example

Haroon-Dweikat-Ntx committed:
restore source example

Haroon-Dweikat-Ntx committed:
deploy pc example

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
Merge branches 'feat/2.1/backup-target', 'feat/2.1/register-unregister', 'feat/2.1/restore-pc' and 'feat/2.1/restore-source' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
protection policies create development

Haroon-Dweikat-Ntx committed:
fix config and network schema

Haroon-Dweikat-Ntx committed:
fix config and network expander

Haroon-Dweikat-Ntx committed:
protection policy development

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/deploy-pc' into feat/2.1/protection-policies

Haroon-Dweikat-Ntx committed:
protection policy info development

Haroon-Dweikat-Ntx committed:
protection policy info development

Haroon-Dweikat-Ntx committed:
promote/restore and get protected resource development

Haroon-Dweikat-Ntx committed:
rename the data source to fetch/list domain manager from pc to domain_managers

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/deploy-pc' into feat/2.1/protection-policies

Haroon-Dweikat-Ntx committed:
rename the data source to fetch/list domain manager from pc to domain_managers

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/deploy-pc' into feat/2.1/protection-policies

Haroon-Dweikat-Ntx committed:
protection policies acc tests

Haroon-Dweikat-Ntx committed:
protection policy updates multiple attributes in the protection policy test case

Haroon-Dweikat-Ntx committed:
protection policy linear retention and auto rollup retention test cases, and add Check Destroy

Haroon-Dweikat-Ntx committed:
protection policy linear retention and auto rollup retention test cases, and add Check Destroy

Haroon-Dweikat-Ntx committed:
protection policy fetch and list data source docs

Haroon-Dweikat-Ntx committed:
protection policy  resource docs

Haroon-Dweikat-Ntx committed:
initialize promote/restore acc tests

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/protection-policies' into feat/2.1/promote-restore-protected-resource

Haroon-Dweikat-Ntx committed:
promote/restore acc tests

Haroon-Dweikat-Ntx committed:
create another provider for second host to promote vm into remote pc

Haroon-Dweikat-Ntx committed:
fix typo and log messages in VG module

Haroon-Dweikat-Ntx committed:
fix typo and log messages in VG module

Haroon-Dweikat-Ntx committed:
implement associate/disassociate category to/from volume group

Haroon-Dweikat-Ntx committed:
merge latest changes from master

Haroon-Dweikat-Ntx committed:
DiffSuppressFunc for category_ids

Haroon-Dweikat-Ntx committed:
acc test case for promote vm, create another instance of provider to automate promote test on diff pc

Haroon-Dweikat-Ntx committed:
acc test case for the promote/restore/get protected source operations, need setup to test it

Haroon-Dweikat-Ntx committed:
backup target info and list

Haroon-Dweikat-Ntx committed:
acc test backup target resource for cluster location and object store location

Haroon-Dweikat-Ntx committed:
kup target info and list test

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
add helper and main_test file

Haroon-Dweikat-Ntx committed:
merge backup target

Haroon-Dweikat-Ntx committed:
initialize restore source test file

Haroon-Dweikat-Ntx committed:
fix backup targets schema

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/backup-target' into feat/2.1/restore-source

Haroon-Dweikat-Ntx committed:
tests for create/delete and get The restore source

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
setup test files

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
test for the list and fetch pc configuration

Haroon-Dweikat-Ntx committed:
fix pc configuration data source module name

Haroon-Dweikat-Ntx committed:
Fix TestAccEraPreCheck Name

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/associate-category-to-vg' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/protection-policies' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/promote-restore-protected-resource' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
Merge remote-tracking branch 'origin/feat/2.1/protection-policies' into feat/2.1/deploy-pc

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
implement dev and test for restorable pcs

Haroon-Dweikat-Ntx committed:
merge deploy-pc, register-unregister, restore-pc, backup-target, restorable-pc, restore-source into prism-v2

Haroon-Dweikat-Ntx committed:
merge deploy-pc, register-unregister, restore-pc, backup-target, restorable-pc, restore-source into prism-v2

Haroon-Dweikat-Ntx committed:
merge deploy-pc, register-unregister, restore-pc, backup-target, restorable-pc, restore-source into prism-v2

Haroon-Dweikat-Ntx committed:
merge deploy-pc, register-unregister, restore-pc, backup-target, restorable-pc, restore-source into prism-v2

Haroon-Dweikat-Ntx committed:
merge deploy-pc, register-unregister, restore-pc, backup-target, restorable-pc, restore-source into prism-v2

Haroon-Dweikat-Ntx committed:
test List Restorable Domain Managers

Haroon-Dweikat-Ntx committed:
test List/Fetch Restore Point

Haroon-Dweikat-Ntx committed:
test Unregister remote cluster

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
remove unnecessary files

Haroon-Dweikat-Ntx committed:
fix backup target bug in get the ext id after creating backup target

Haroon-Dweikat-Ntx committed:
fix backup target bug in get the ext id after creating backup target

Haroon-Dweikat-Ntx committed:
list restore points test

Haroon-Dweikat-Ntx committed:
unregister pc-pc test

Haroon-Dweikat-Ntx committed:
test fetch restore point

Haroon-Dweikat-Ntx committed:
acc test for deploy PC

Haroon-Dweikat-Ntx committed:
backup target update test

Haroon-Dweikat-Ntx committed:
acc test restore pc

Haroon-Dweikat-Ntx committed:
acc test restore pc

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
add docs & examples

Haroon-Dweikat-Ntx committed:
prismv2 fixes

Haroon-Dweikat-Ntx committed:
update test variable data_protection to availability_zone

Haroon-Dweikat-Ntx committed:
update powerOffPC func to choose the correct vm pc

Haroon-Dweikat-Ntx committed:
update restore pc examples

Haroon-Dweikat-Ntx committed:
update restore pc test case

Haroon-Dweikat-Ntx committed:
update backup target and restore source examples

Haroon-Dweikat-Ntx committed:
update restore pc test

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
fix restore pc test

Haroon-Dweikat-Ntx committed:
fix restore pc test

Haroon-Dweikat-Ntx committed:
fix restore source, if the restore source auto-deleted it will be created again in the same resource

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
restore pc example fix

Haroon-Dweikat-Ntx committed:
fix restore source example

Haroon-Dweikat-Ntx committed:
refactor prism test cases, skip deploy pc and restore pc tests

GullapalliAkhil commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
 			"nutanix_storage_container_stats_info_v2":         storagecontainersv2.DatasourceNutanixStorageStatsInfoV2(),
 			"nutanix_category_v2":                             prismv2.DatasourceNutanixCategoryV2(),
 			"nutanix_categories_v2":                           prismv2.DatasourceNutanixCategoriesV2(),
+			"nutanix_pc_v2":                                   prismv2.DatasourceNutanixFetchPcV2(),
+			"nutanix_pcs_v2":                                  prismv2.DatasourceNutanixListPcsV2(),
```
IMHO: The naming here doesn't convey what exactly it does, It'll be really helpful if we have added a better naming.
I have a thought, can't we add.
nutanix_get_pc_v2, nutanix_list_pcs_v2?

Haroon-Dweikat-Ntx replied:
data sources for list and get data only, so nutanix_pc_v2 for getting pc details and nutanix_pcs_v2 for listing PC 
that's the used name pattern in the provider

Haroon-Dweikat-Ntx committed:
lint fixes

Haroon-Dweikat-Ntx committed:
fix log messages and task var names

Haroon-Dweikat-Ntx committed:
remove blank line

Haroon-Dweikat-Ntx committed:
fix examples and docs

bhati-pradeep commented on examples/backup_target_v2/main.tf:
```diff
@@ -0,0 +1,87 @@
+  ]
+}
+
+// using cluster_location
```
```suggestion
// Create a backup target to cluster location to take PC backups
```

bhati-pradeep commented on examples/backup_target_v2/main.tf:
```diff
@@ -0,0 +1,87 @@
+
+
+
+//using object store location
```
Elaborate more in example

bhati-pradeep commented on examples/backup_target_v2/main.tf:
```diff
@@ -0,0 +1,87 @@
+    }
+  }
+  lifecycle {
+    ignore_changes = [
```
Comment why we have added ignore changes

bhati-pradeep commented on examples/backup_target_v2/main.tf:
```diff
@@ -0,0 +1,87 @@
+  domain_manager_ext_id = local.domainManagerExtID
+}
+
+// get backup target
```
```suggestion
// get backup target created above 
```

bhati-pradeep commented on examples/restorable_pcs_v2/main.tf:
```diff
@@ -0,0 +1,73 @@
+terraform {
```
Add summary steps in comments what you will be doing in this script.
1. Fetch PC external IDs
2. Create backup target to cluster location for storing backups
3. Create a restore source to a cluster location for restoring PC from that cluster.

bhati-pradeep commented on examples/restore_pc_v2/main.tf:
```diff
@@ -0,0 +1,177 @@
+  }
+}
+
+#defining nutanix configuration
```
```suggestion
# Define provider configuration to connect to Nutanix PC 
```

bhati-pradeep commented on examples/restore_pc_v2/main.tf:
```diff
@@ -0,0 +1,177 @@
+  insecure = true
+}
+
+# define another alias for the provider,  PE
```
```suggestion
# Define another alias provider configuration to connect to Nutanix PE 
```

bhati-pradeep commented on examples/restore_pc_v2/main.tf:
```diff
@@ -0,0 +1,177 @@
+    }
+  }
+}
+
```
Add comments for explaining what each step does

bhati-pradeep commented on examples/restore_pc_v2/main.tf:
```diff
@@ -0,0 +1,177 @@
+    }
+  }
+
+  # after restore pc, you need to reset the password of the pc_user
```
This is a hack. No need. Please remove.

bhati-pradeep commented on examples/restore_point_v2/main.tf:
```diff
@@ -0,0 +1,91 @@
+}
+
+# get restore point details
+data "nutanix_restore_point_v2" "test" {
```
Please provide unique and relavant names to resource or data soure blocks

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
+			"nutanix_pc_v2":                                   prismv2.DatasourceNutanixFetchPcV2(),
+			"nutanix_pcs_v2":                                  prismv2.DatasourceNutanixListPcsV2(),
+			"nutanix_restorable_pcs_v2":                       prismv2.DatasourceNutanixListRestorablePcsV2(),
+			"nutanix_restore_points_v2":                       prismv2.DatasourceNutanixFetchRestorePointsV2(),
```
```suggestion
			"nutanix_pc_restore_points_v2":                       prismv2.DatasourceNutanixFetchRestorePointsV2(),
```

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
+			"nutanix_pcs_v2":                                  prismv2.DatasourceNutanixListPcsV2(),
+			"nutanix_restorable_pcs_v2":                       prismv2.DatasourceNutanixListRestorablePcsV2(),
+			"nutanix_restore_points_v2":                       prismv2.DatasourceNutanixFetchRestorePointsV2(),
+			"nutanix_restore_point_v2":                        prismv2.DatasourceNutanixFetchRestorePointV2(),
```
```suggestion
			"nutanix_pc_restore_point_v2":                        prismv2.DatasourceNutanixFetchRestorePointV2(),
```

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
+			"nutanix_restorable_pcs_v2":                       prismv2.DatasourceNutanixListRestorablePcsV2(),
+			"nutanix_restore_points_v2":                       prismv2.DatasourceNutanixFetchRestorePointsV2(),
+			"nutanix_restore_point_v2":                        prismv2.DatasourceNutanixFetchRestorePointV2(),
+			"nutanix_backup_target_v2":                        prismv2.DatasourceNutanixBackupTargetV2(),
```
```suggestion
			"nutanix_pc_backup_target_v2":                        prismv2.DatasourceNutanixBackupTargetV2(),
```

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
+			"nutanix_restore_points_v2":                       prismv2.DatasourceNutanixFetchRestorePointsV2(),
+			"nutanix_restore_point_v2":                        prismv2.DatasourceNutanixFetchRestorePointV2(),
+			"nutanix_backup_target_v2":                        prismv2.DatasourceNutanixBackupTargetV2(),
+			"nutanix_backup_targets_v2":                       prismv2.DatasourceNutanixBackupTargetsV2(),
```
```suggestion
			"nutanix_pc_backup_targets_v2":                       prismv2.DatasourceNutanixBackupTargetsV2(),
```

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -267,6 +267,14 @@ func Provider() *schema.Provider {
+			"nutanix_restore_point_v2":                        prismv2.DatasourceNutanixFetchRestorePointV2(),
+			"nutanix_backup_target_v2":                        prismv2.DatasourceNutanixBackupTargetV2(),
+			"nutanix_backup_targets_v2":                       prismv2.DatasourceNutanixBackupTargetsV2(),
+			"nutanix_restore_source_v2":                       prismv2.DatasourceNutanixRestoreSourceV2(),
```
```suggestion
			"nutanix_pc_restore_source_v2":                       prismv2.DatasourceNutanixRestoreSourceV2(),
```

bhati-pradeep commented on nutanix/provider/provider.go:
```diff
@@ -357,6 +365,12 @@ func Provider() *schema.Provider {
 			"nutanix_saml_identity_providers_v2":              iamv2.ResourceNutanixSamlIdpV2(),
 			"nutanix_storage_containers_v2":                   storagecontainersv2.ResourceNutanixStorageContainersV2(),
 			"nutanix_category_v2":                             prismv2.ResourceNutanixCategoriesV2(),
+			"nutanix_deploy_pc_v2":                            prismv2.ResourceNutanixDeployPcV2(),
```
nutanix_pc_deploy_v2
nutanix_pc_backup_targets_v2
nutanix_pc_restore_source_v2
nutanix_pc_restore_v2
nutanix_pc_unregistration_v2
nutanix_pc_registration_v2

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_backup_target_v2_test.go:
```diff
@@ -0,0 +1,96 @@
+			// Create backup target, cluster location
+			{
+				Config: testAccFetchBackupTargetDatasourceConfig(),
+				Check: resource.ComposeTestCheckFunc(
```
@Haroon-Dweikat-Ntx Please have comparison checks as well with expected values its correct value or not rather than just checking if values are just set.

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_backup_targets_v2_test.go:
```diff
@@ -0,0 +1,35 @@
+			{
+				Config: testAccListBackupTargetsDatasourceConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(datasourceNameBackupTargets, "backup_targets.0.ext_id"),
```
@Haroon-Dweikat-Ntx Please have comparison checks as well with expected values its correct value or not rather than just checking if values are just set.

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_backup_target_v2_test.go:
```diff
@@ -0,0 +1,96 @@
+
+const datasourceNameBackupTarget = "data.nutanix_backup_target_v2.test"
+
+func TestAccV2NutanixBackupTargetDatasource_Basic(t *testing.T) {
```
@Haroon-Dweikat-Ntx Please have comparison checks as well with expected values its correct value or not rather than just checking if values are just set.

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_backup_targets_v2_test.go:
```diff
@@ -0,0 +1,35 @@
+				),
+			},
+			// List backup targets
+			{
```
@Haroon-Dweikat-Ntx Please have comparison checks as well with expected values its correct value or not rather than just checking if values are just set.

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_pc_v2.go:
```diff
@@ -0,0 +1,550 @@
+		if pcConfig.BootstrapConfig != nil {
+			pcConfigMap["bootstrap_config"] = flattenBootstrapConfig(pcConfig.BootstrapConfig)
+		}
+		//if pcConfig.Credentials != nil {
```
Remove if not needed

bhati-pradeep commented on nutanix/services/prismv2/data_source_nutanix_restore_point_v2_test.go:
```diff
@@ -0,0 +1,98 @@
+					fmt.Printf("Step 4: List Restore Points\n")
+				},
+				Config: testRestoreSourceConfig() + testAccListRestorePointsConfig(),
+				Check: resource.ComposeTestCheckFunc(
```
@Haroon-Dweikat-Ntx Please have comparison checks as well with expected values its correct value or not rather than just checking if values are just set.

Haroon-Dweikat-Ntx replied:
List the restore points. They are auto-generated, so we don't have a specific extID to verify.
Instead, we simply check if the extID is set, as the API returns it and we store it in the state file.If the API fails to return the extID, the test will fail.

bhati-pradeep commented on nutanix/services/prismv2/helper.go:
```diff
@@ -0,0 +1,82 @@
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+// funcs
```
Please add a one line description for this function about what its doing

bhati-pradeep commented on nutanix/services/prismv2/helper_test.go:
```diff
@@ -0,0 +1,801 @@
+package prismv2_test
```
Please add code comments and one liner description on each function on what each function does and its usecase

bhati-pradeep commented on nutanix/services/prismv2/main_test.go:
```diff
@@ -0,0 +1,66 @@
+
+var (
+	path, _  = os.Getwd()
+	filepath = path + "/../../../test_config_v2.json"
```
Seems its not used anywhere. Why do we need this ?

Haroon-Dweikat-Ntx replied:
no, its used in resource tests (backup target, restore source, pc deploy ..)

bhati-pradeep commented on nutanix/services/prismv2/resource_nutanix_backup_target_v2_test.go:
```diff
@@ -0,0 +1,268 @@
+  }
+}
+
+`, filepath)
```
where is this file path coming from ?

Haroon-Dweikat-Ntx replied:
From main_test.go, we define it as a global variable shared by all tests.

bhati-pradeep commented on nutanix/services/prismv2/resource_nutanix_deploy_pc_v2.go:
```diff
@@ -0,0 +1,1135 @@
+							Optional: true,
+							Computed: true,
+							MinItems: 1,
+							MaxItems: 3, //nolint:gomnd
```
Is this limit by API ?

Haroon-Dweikat-Ntx replied:
Yes, these validations follow the guidelines outlined in the API documentation.

bhati-pradeep commented on nutanix/services/prismv2/resource_nutanix_restore_pc_v2.go:
```diff
@@ -0,0 +1,160 @@
+	return nil
+}
+
+func ResourceNutanixRestorePcUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
If its a action module then we should be doing triggering restore PC in every run ? Whats the expectation ?

bhati-pradeep commented on nutanix/services/prismv2/resource_nutanix_restore_source_v2.go:
```diff
@@ -0,0 +1,274 @@
+	return nil
+}
+
+func ResourceNutanixRestoreSourceV2Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Please mention this all flows in docs

bhati-pradeep commented on nutanix/services/vmmv2/data_source_nutanix_virtual_machines_v2.go:
```diff
@@ -1296,16 +1295,13 @@ func DatasourceNutanixVirtualMachinesV4Read(ctx context.Context, d *schema.Resou
-		errorList := data["error"].([]interface{})
-		errorMessage := errorList[0].(map[string]interface{})
-		return diag.Errorf("error while fetching vms : %v", errorMessage["message"])
+		return diag.Errorf("error while fetching vms : %v", err)
```
why this change is required ?

Haroon-Dweikat-Ntx replied:
For data sources listing elements, I discovered that if the API returns an empty list (e.g., when filtering by name and no corresponding VM exists), it used to trigger a nil pointer exception. Now, it returns an error message instead.

Haroon-Dweikat-Ntx committed:
rename modules:
- nutanix_pc_deploy_v2
- nutanix_pc_backup_targets_v2
- nutanix_pc_restore_source_v2
- nutanix_pc_restore_v2
- nutanix_pc_unregistration_v2
- nutanix_pc_registration_v2

fix examples and docs
add mor comment

Haroon-Dweikat-Ntx committed:
lint fix

Haroon-Dweikat-Ntx committed:
change Update action call create action for pc deploy and pc  restore

Haroon-Dweikat-Ntx committed:
rename examples for prism modules

Haroon-Dweikat-Ntx committed:
pc restore resource update should return nil

Haroon-Dweikat-Ntx committed:
update docs for pc restore/ pc deploy and pc restore source

bhati-pradeep commented:
Resolve conflict and land

Haroon-Dweikat-Ntx committed:
Merge branch 'release/2.1' into feat/2.1/prism-v2

Haroon-Dweikat-Ntx committed:
lint fix
